pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol

```diff
@@ -0,0 +1,201 @@
+     */
+    function getTotalSupplyPoint() external view returns (IVotingEscrow.Point memory) {
+        IVotingEscrow votingEscrow = getVotingEscrow();
+        uint256 totalSupplyEpoch = votingEscrow.epoch();
+        return votingEscrow.point_history(totalSupplyEpoch);
+    }
+
+    /**
+     * @notice Get timestamp when `user`'s lock finishes.
+     * @dev The returned value is taken directly from the voting escrow.
+     */
+    function getLockedEnd(address user) external view returns (uint256) {
+        return getVotingEscrow().locked__end(user);
+    }
+
+    /**
+     * @notice Returns the local user corresponding to an address on a remote chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given remote user.
+     * @param remoteUser - Address of the user on the remote chain which are querying the local address for.
+     * @param chainId - The chain ID of the network which this user is on.
+     */
+    function getLocalUser(address remoteUser, uint256 chainId) public view returns (address) {
+        return _remoteToLocalAddressMap[chainId][remoteUser];
+    }
+
+    /**
+     * @notice Returns the remote user corresponding to an address on the local chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given local user.
+     * @param localUser - Address of the user on the local chain which are querying the remote address for.
+     * @param chainId - The chain ID of the network which the remote user is on.
+     */
+    function getRemoteUser(address localUser, uint256 chainId) public view returns (address) {
+        return _localToRemoteAddressMap[chainId][localUser];
+    }
+
+    function getRemappingManager(address localUser) public view returns (address) {
+        return _localRemappingManager[localUser];
+    }
+
+    // Remapping Setters
+
+    /**
+     * @notice Sets up a mapping from `localUser`'s veBAL balance to `remoteUser` for chain `chainId`.
+     * @dev In order to set up a remapping on this contract, `localUser` must be a smart contract which has been
+     * allowlisted to hold veBAL. EOAs are expected to set up any delegation of their veBAL on L2s directly.
+     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
+     * @param remoteUser - The address to receive `localUser`'s balance of veBAL on the remote chain.
+     * @param chainId - The chain id of the remote chain on which `remoteUser` resides.
+     */
+    function setNetworkRemapping(
+        address localUser,
+        address remoteUser,
+        uint256 chainId
+    ) external {
+        _require(msg.sender == localUser || msg.sender == _localRemappingManager[localUser], Errors.SENDER_NOT_ALLOWED);
+        require(_isAllowedContract(localUser), "Only contracts which can hold veBAL can set up a mapping");
+        require(remoteUser != address(0), "Zero address cannot be used as remote user");
```

Comment:
How can a remapping be cleared then? It can be changed, but not cleared unless the contract stops passing the smart wallet check.

Reply:
The 'clear' feature is to deter griefing attacks.

If an account was misbehaving and mapping to somebody else's address, it can be deny-listed in the wallet checker, and then anyone can remove the mapping.

If the user wants to clear its own rightful mapping they can always do so.